code/modules/projectiles/gun_system.dm

```diff
@@ -687,7 +715,7 @@ and you're good to go.
 
 
 	play_fire_sound(gun_user)
-	muzzle_flash(firing_angle, gun_user)
+	muzzle_flash(firing_angle, loc)
```

Comment:
hum, is this not breaking muzze_flash of normal guns?

Reply:
nope, cuz gun_user = loc.